src/transformers/models/dpt/modeling_dpt.py

```diff
@@ -0,0 +1,1127 @@
+        ]
+        self.convs = nn.ModuleList()
+        for channel in self.post_process_channels:
+            self.convs.append(nn.Conv2d(channel, config.channels, kernel_size=3, padding=1, bias=False))
+
+        # fusion
+        self.fusion_blocks = nn.ModuleList()
+        for _ in range(len(self.convs)):
+            self.fusion_blocks.append(DPTFeatureFusionBlock(config))
+        self.fusion_blocks[0].res_conv_unit1 = None  # not sure why this is done in mmseg
+
+    def forward(self, hidden_states: List[torch.Tensor]) -> List[torch.Tensor]:
+        if len(hidden_states) != len(self.post_process_channels):
+            raise ValueError("The number of hidden states should be equal to the number of post-process channels.")
+
+        # postprocess hidden states
+        features = self.reassemble_blocks(hidden_states)
+
+        features = [self.convs[i](feature) for i, feature in enumerate(features)]
+
+        # fusion blocks
+        output = []
+        for i in range(len(self.fusion_blocks)):
+            if i == 0:
+                out = self.fusion_blocks[i](features[-1])
+            else:
+                out = self.fusion_blocks[i](out, features[-(i + 1)])
+            output.append(out)
+
+        return output
+
+
+class DPTInterpolate(nn.Module):
+    """Interpolation module."""
+
+    def __init__(self, scale_factor, mode, align_corners=False):
+        """Init.
+        Args:
+            scale_factor (float): scaling
+            mode (str): interpolation mode
+        """
+        super().__init__()
+
+        self.interpolate = nn.functional.interpolate
+        self.scale_factor = scale_factor
+        self.mode = mode
+        self.align_corners = align_corners
+
+    def forward(self, x):
+        """Forward pass.
+        Args:
+            x (tensor): input
+        Returns:
+            tensor: interpolated data
+        """
+
+        x = self.interpolate(x, scale_factor=self.scale_factor, mode=self.mode, align_corners=self.align_corners)
+
+        return x
+
+
+class DPTDepthEstimationHead(nn.Module):
+    """
+    Output head head consisting of 3 convolutional layers. It progressively halves the feature dimension and upsamples
+    the predictions to the input resolution after the first convolutional layer (details can be found in the paper's
+    supplementary material).
+    """
+
+    def __init__(self, config):
+        super().__init__()
+
+        self.config = config
+
+        features = config.channels
+        self.head = nn.Sequential(
+            nn.Conv2d(features, features // 2, kernel_size=3, stride=1, padding=1),
+            DPTInterpolate(scale_factor=2, mode="bilinear", align_corners=True),
+            nn.Conv2d(features // 2, 32, kernel_size=3, stride=1, padding=1),
+            nn.ReLU(True),
+            nn.Conv2d(32, 1, kernel_size=1, stride=1, padding=0),
+            nn.ReLU(True),
+            nn.Identity(),
+        )
```

Comment:
why are we using `ReLU` `inplace` here?

Reply:
Perhaps because it's in an `nn.Sequential`?

Reply:
Maybe to save memory?